core/summary_api.php

```diff
@@ -1363,23 +1363,31 @@ function summary_by_dates_bug_count( array $p_date_array, array $p_filter = null
 		$t_count_array['open'][$t_ix] = 0;
 		$t_count_array['close'][$t_ix] = 0;
 	}
-	# count is accumulated, and date ranges are ordered in the result
-	$t_count_open = 0;
-	$t_count_closed = 0;
+	# The query returns the count specific to each date range (some ranges may
+	# not exists in the query result if the count is 0)
+	# Fill the array with those values first.
```

Comment:
I'm confused by this comment, it almost seems that it is referring to the initialization code above.

Reply:
It's a 3-step process
1. initialize $t_count_array to zeros
2. update counts for days based on query results
3. accumulate the counters